sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -139,6 +140,13 @@ case class HiveTableScanExec(
   }
 
   protected override def doExecute(): RDD[InternalRow] = {
+    val locationPath = new Path(relation.catalogTable.location)
+    val fs = locationPath.getFileSystem(sparkSession.sessionState.newHadoopConf())
+
+    // if the table location is not exists, return an empty RDD
+    if (!fs.exists(locationPath)) {
+      return new EmptyRDD[InternalRow](sparkSession.sparkContext)
```

Comment:
I do it here for both non-partition table and partition table，while the partition table run well when the location does not exist with verifyPartitionPath set true，If we also want to run well when verifyPartitionPath set false， we should also do it in makeRDDForPartitionTable， then under this situation I do it here for both non-partition table and partition table.